File: insertion.c
void insertionSort(int *data,int length){
	int current;

	current=1;

	while(current<length) {
		int index;
		int placeFound;

		index=current;
		placeFound=0;

		while(placeFound<index) {
			int temp;
			int i;

			if(data[0]>=data[index]) {
				for(i=0;i<index;i++) {
					temp=data[index-i];
					data[index-i]=data[index-i-1];
					data[index-i-1]=temp;
				}
			}
			else if(data[placeFound]<=data[index] && data[placeFound+1]>=data[index]) {
				for(i=0;i<index-(placeFound+1);i++) {
					temp=data[index-i];
					data[index-i]=data[index-i-1];
					data[index-i-1]=temp;
				}
			}
			placeFound++;
		}
		current++;
	}
}

File: insertion.py
def insertionSort(data,length):
    current=1;
    while(current<length):
        index=current;
        placeFound=0;
        while(index>0 and not placeFound):
            if(data[index]<data[index-1]):
               data[index],data[index-1]=data[index-1],data[index]
               index-=1
            else:
               placeFound=1
        current+=1

